### ISSUE TITLE
Navigation search bar needs accessibility improvements

### ISSUE THREAD

**Issue by amberhinds** [tomusborne/generatepress#430]
### Description
There are a number of accessibility problems with the search form in the navigation menu. Specifically, I am referencing this code for the open/close button:
`<div class="menu-bar-items">
     <span class="menu-bar-item search-item active close-search">
          <a aria-label="Close Search Bar" href="#">
               <span class="gp-icon icon-search">
                    <svg viewBox="0 0 512 512" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" width="1em" height="1em"><path fill-rule="evenodd" clip-rule="evenodd" d="M208 48c-88.366 0-160 71.634-160 160s71.634 160 160 160 160-71.634 160-160S296.366 48 208 48zM0 208C0 93.125 93.125 0 208 0s208 93.125 208 208c0 48.741-16.765 93.566-44.843 129.024l133.826 134.018c9.366 9.379 9.355 24.575-.025 33.941-9.379 9.366-24.575 9.355-33.941-.025L337.238 370.987C301.747 399.167 256.839 416 208 416 93.125 416 0 322.875 0 208z"></path></svg><svg viewBox="0 0 512 512" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" width="1em" height="1em"><path d="M71.029 71.029c9.373-9.372 24.569-9.372 33.942 0L256 222.059l151.029-151.03c9.373-9.372 24.569-9.372 33.942 0 9.372 9.373 9.372 24.569 0 33.942L289.941 256l151.03 151.029c9.372 9.373 9.372 24.569 0 33.942-9.373 9.372-24.569 9.372-33.942 0L256 289.941l-151.029 151.03c-9.373 9.372-24.569 9.372-33.942 0-9.372-9.373-9.372-24.569 0-33.942L222.059 256 71.029 104.971c-9.372-9.373-9.372-24.569 0-33.942z"></path></svg>
               </span>
          </a>
     </span>
</div>`

and this code for the search form itself.
`<form method="get" class="search-form navigation-search nav-search-active" action="https://gpsites.co/marketer/"> 
     <input type="search" class="search-field" value="" name="s" title="Search">
</form>`

The code examples here have been pulled from your [Marketer demo](https://gpsites.co/marketer/) in your Site Library.

Here are the specific problems:

1. The open/close buttons are incorrectly tagged as a link when they should actually use a `<button>` tag. 
   - Screen readers handle links slightly differently than they do buttons. All links and buttons are tab-able, but pressing the Space bar or Enter/Return triggers a button, whereas pressing the Enter/Return key only triggers a link.
   - Links are used to navigate to a new page or a different page of the same page, not to trigger an interaction or open/close a modal as we're seeing with this search form. 
   - Ideally, this should be re-coded to have a `<button>` tag, but if you're concerned that will be a breaking change, then it could be resolved by adding `role="button"` to the existing `<a>`.
2. Input field is missing a visible label.
   - Missing labels on form fields is a WCAG violation. It is also important that the label is visible (and not be visually hidden but available to screen reader users), because sighted users benefit from the label too.
   - The label needs to stay visible even when content has been entered in the field so that the field's purpose can be identified easily without having to delete the content in the field (I.e., placeholder text alone is not sufficient).

I hope this is helpful. I appreciate you resolving these accessibility problems. please let me know if you have any questions.

---

#### References
Here are resources that might be helpful.

1. [ARIA button role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role)
2. [W3C Labeling Controls](https://www.w3.org/WAI/tutorials/forms/labels/)

**Comment by amberhinds** [comment]
I have updated the issue above with a few more details for problems that were identified during user testing. Are you able to provide a timeline for this fix?

**Comment by tomusborne** [comment]
@amberhinds Thanks so much for this - really helpful! We're in the process of re-thinking our Navigation Search feature and may end up deprecating the current version of it in favor of a newly thought-out version.

I'm going to tag this as 3.3.0 so we can refer here as we build out the new feature.

**Comment by amberhinds** [comment]
Hi @tomusborne, thanks for the message and for taking my feedback into consideration. I'm happy to test the new version whenever you have it ready.

**Comment by SchneiderSam** [comment]
> We're in the process of re-thinking our Navigation Search feature and may end up deprecating the current version of it in favor of a newly thought-out version.

@tomusborne  The Gutenberg-Way?

**Comment by tomusborne** [comment]
We've added a new Navigation Search Modal feature to GP 3.3.0: https://generatepress.com/a-new-navigation-search/

Would love any feedback on its accessibility.

**Comment by amberhinds** [comment]
Hey Tom,

I can test it if you want, but do you have a zip or - even better - a link
to a demo site?

*Amber Hinds*
Creative Director & CEO
Road Warrior Creative

***@***.*** • (512) 942-5858 <5129425858> ext. 3
*Connect With Me on LinkedIn <http://bit.ly/AmberLinkedIn> *•
https://roadwarriorcreative.com

**Comment by amberhinds** [comment]
Thanks, but I don't have GeneratePress. 🙂

*Amber Hinds*
Creative Director & CEO
Road Warrior Creative

***@***.*** • (512) 942-5858 <5129425858> ext. 3
*Connect With Me on LinkedIn <http://bit.ly/AmberLinkedIn> *•
https://roadwarriorcreative.com

**Comment by Studio88-KM** [comment]
Amber, just emailed you. You can use mine if you'd like.
